packages/squiggle-lang/src/value/index.ts

```diff
@@ -78,6 +79,23 @@ class VArray extends BaseValue implements Indexable {
       )
     );
   }
+
+  shuffle() {
+    return new VArray(shuffle(this.value));
+  }
+
+  isEqual(other: VArray, compareValues: (a: Value, b: Value) => boolean) {
```

Comment:
It's possible it's not needed. The reason I used it was because of dependency issues  - vArray was declared before Value, but here we want to compare values. This function isn't meant to be called directly - instead, one does isEqual on Value, which then calls this with that param.

I guess, if there's no better solution, one option here is to have this be a protected class, or change it to "_isEqual"

Reply:
Huh. I don't really see the problem with calling `isEqual` directly, it looks like a correct API for an untyped "compare these two values" to me.

The pattern with `a._isEqual(b: Value)` that you used on `BaseDist` is ok, but it's less strictly typed, and more verbose.

I keep thinking that there must be a better solution with generics, but don't have one yet.